### PR TITLE
BUG/ENH/DOC: spatial: Better handle & document `Rotation` splitting

### LINKED ISSUE #21351: ENH: spatial: Rotation: add `split` to break apart object with multiple values into list of rotations
### Is your feature request related to a problem? Please describe.

This is essentially the opposite of `Rotation.concatenate()` and I think it should mirror the syntax of `numpy.split()` https://numpy.org/doc/stable/reference/generated/numpy.split.html

My current use case is making a pandas dataframe column with Rotations, which requires turning it into a list.

Should be pretty easy to implement with just calling `numpy.split()` under the hood, but want to get some feedback before opening an MR.

### Describe the solution you'd like.

Current `numpy.split()` behavior:
```python
import numpy as np

a = np.array([1, 2, 3, 4])
np.split(a, len(a))
# returns [array([1]), array([2]), array([3]), array([4])]

np.split(a, [1, 2])
# returns [array([1]), array([2]), array([3, 4])]
```

New similar `Rotation.split()` behavior:
```python
from scipy.spatial.transform import Rotation as R

r1 = R.from_rotvec([0, 0, 1])
r2 = R.from_rotvec([0, 0, 2])
r3 = R.from_rotvec([0, 0, 3])
r4 = R.from_rotvec([0, 0, 4])
r34 = R.concatenate([r3, r4])
rc = R.concatenate([r1, r2, r3, r4])

rc.split(len(rc))
# returns [r1, r2, r3, r4]

rc.split([1, 2])
# returns [r1, r2, r34]

# This equality would always hold, no matter what was in *
rc == R.concatenate(rc.split(*))
```

### Describe alternatives you've considered.

The following works, but is clunky and less flexible. This is pretty much exactly what would be happening under the hood.
```python
[R.from_quat(np.squeeze(qs)) for qs in np.split(rc.as_quat(), len(rc))]
# returns [r1, r2, r3, r4]

[R.from_quat(np.squeeze(qs)) for qs in np.split(rc.as_quat(), [1, 2])]
# returns [r1, r2, r34]
```

### Additional context (e.g. screenshots, GIFs)

_No response_

### PR DESCRIPTION
**Pull request by scottshambaugh**
#### Reference issue
Closes https://github.com/scipy/scipy/issues/21351

Replaces https://github.com/scipy/scipy/pull/21494

#### What does this implement/fix?
This has three changes:
1) BUG: Disallow creating rotations from empty size (0, 4) arrays (was silently breaking things)
2) ENH: `Rotation.concatenate` now accepts a bare `Rotation` object, and will return a copy of it
3) DOC: Added information to docs for `Rotation.concatenate` and `Rotation.__getitem__` to better show how to split rotations via indexing and list comprehension

#### Additional information
I'm happy to split this into multiple MRs if desired, but they're pretty small changes so I figured this might be easiest. The commits are split into one for each change.

### CODE REVIEW

File: scipy/spatial/transform/tests/test_rotation.py [hunk 1]
```diff
@@ -1758,6 +1769,14 @@ def test_concatenate():
     result = Rotation.concatenate(split)
     assert_equal(rotation.as_quat(), result.as_quat())
 
+    # Test Rotation input for multiple rotations
+    result = Rotation.concatenate(rotation)
```
Comment (j-bowhay) [inline scipy/spatial/transform/tests/test_rotation.py:1773]: Can you also test that the result is a copy?

Comment (scottshambaugh) [reply scipy/spatial/transform/tests/test_rotation.py:1773]: Sure thing, done

**Review by j-bowhay — COMMENTED** [review]
Thanks, test failure looks really but otherwise looks in good shape

**Review by j-bowhay — APPROVED** [review]
LGTM

**Pull request merged**